### PR TITLE
Add RFS and MCFS

### PR DESCRIPTION
**Pull request by MadHypnofrog**

### CODE REVIEW

File: ITMO_FS/utils/functions.py [hunk 1]
```diff
@@ -74,3 +73,42 @@ def knn(X, y, index, k, allClasses=True):
                     [(i, x) for i, x in enumerate(X) if i != index and (allClasses or y[i] == y[index])])
     nearest = sorted(distances, key=lambda x: x[1])[:k]
     return np.array(list(map(lambda x: x[0], nearest)))
+
+def matrix_norm(M):
```
[the remark this answers is not in the document]
Comment (MadHypnofrog) [reply ITMO_FS/utils/functions.py:77]: it's possible to replace matrix_norm with np.linalg.norm(M, axis=1) but overall these are just wrappers added for convenience

File: ITMO_FS/filters/sparse/NDFS.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+		S = func(indices)
+
+		A = np.diag(S.sum(axis=0))
+		L = power_neg_half(A).dot(A - S).dot(power_neg_half(A))
+		
+		if y != None:
+			if len(y.shape) == 2:
+				Y = y
+			else:
+				Y = OneHotEncoder().fit_transform(y.reshape(-1, 1)).toarray()
```
[the remark this answers is not in the document]
Comment (MadHypnofrog) [reply ITMO_FS/filters/sparse/NDFS.py:106]: the original paper assumes that Y is a one-hot encoding of classes; here it is possible to pass the original array as well so that the algorithm would convert it to OHE
